plugins/CoreHome/Controller.php

```diff
@@ -300,4 +300,11 @@ public function saveViewDataTableParameters()
 
         ViewDataTableManager::saveViewDataTableParameters($login, $reportId, $parameters);
     }
+
+    public function getDateString()
```

Comment:
This would need to be an API method, and would also need to check for permissions.

Reply:
Like `Piwik::checkUserHasSomeViewAccess();`

Reply:
Hi @tsteur, good catch on the missing function-level access control! I can add this permission check.

What does it mean, this needs to be an API method? Does it need to get documented somewhere, or some code changed?

Reply:
It would be documented kind of automatically once it is an API method. As suggested below, I would maybe remove all changes in PHP and instead only add the date in JS to the title (always). This would remove a lot of problems and complexity plus ensure consistent behaviour.